book/scheduling.md

```diff
@@ -1219,15 +1203,40 @@ class Browser:
 
 Do the same with any other calls from the `Browser` to the `Tab`.
 
-Communication in the other direction is a little subtler. We already
-have the `set_needs_animation_frame` method, but now we need the `Tab`
-to call `commit` when it's finished creating a display list.
+So now we have the browser thread telling the main thread what to do
+Communication in the other direction is a little subtler.
 
-If you look carefully at our raster-and-draw code, you'll see that to
-draw a display list we also need to know the URL (to update the
-browser chrome), the document height (to allocate a surface of the
-right size), and the scroll position (to draw the right part of the
-surface). Let's make a simple class for storing this data:
+::: {.further}
+
+Originally, threads were a mechanism for improving *responsiveness*
+via pre-emptive multitasking, not *throughput* (frames per second).
```

Comment:
[^gil-second]: And our browser still doesn't do that, due to the GIL, lolz.

Reply:
?

Reply:
Nevermind, not important.